lib/types.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2016 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rkt
+
+import "github.com/coreos/rkt/networking/netinfo"
+
+// AppState defines the state of the app.
+type AppState string
+
+const (
+	AppStateUnknown AppState = "unknown"
+	AppStateCreated AppState = "created"
+	AppStateRunning AppState = "running"
+	AppStateExited  AppState = "exited"
+)
+
+type (
+	// Mount defines the mount point.
+	Mount struct {
+		// Name of the mount.
+		Name string `json:"name"`
+		// Container path of the mount.
+		ContainerPath string `json:"container_path"`
+		// Host path of the mount.
+		HostPath string `json:"host_path"`
+		// Whether the mount is read-only.
+		ReadOnly bool `json:"read_only"`
+		// TODO(yifan): What about 'SelinuxRelabel bool'?
+	}
+
+	// App defines the app object.
+	App struct {
+		// Name of the app.
+		Name string `json:"name"`
+		// State of the app, can be created, running, exited, or unknown.
+		State AppState `json:"state"`
+		// Creation time of the container, nanoseconds since epoch.
+		CreatedAt *int64 `json:"created_at,omitempty"`
+		// Start time of the container, nanoseconds since epoch.
+		StartedAt *int64 `json:"started_at,omitempty"`
+		// Finish time of the container, nanoseconds since epoch.
+		FinishedAt *int64 `json:"finished_at,omitempty"`
+		// Exit code of the container.
+		ExitCode *int `json:"exit_code,omitempty"`
+		// Image ID of the container.
+		ImageID string `json:"image_id"`
+		// Mount points of the container.
+		Mounts []*Mount `json:"mounts,omitempty"`
+		// Annotations of the container.
+		Annotations map[string]string `json:"annotations,omitempty"`
+	}
+
+	// Pod defines the pod object.
+	Pod struct {
+		UUID string `json:"name"`
+		// State is defined in pkg/pod/pods.go
+		State    string            `json:"state"`
+		Networks []netinfo.NetInfo `json:"networks,omitempty"`
+		// TODO(yifan): Decide if we want to include detailed app info.
```

Comment:
I guess it is preferable to avoid duplicating app info, otherwise when mutating a pod we will have more bookeeping to do in order to ensure everything is in sync.